server/src/test/java/org/elasticsearch/index/shard/RefreshListenersTests.java

```diff
@@ -503,8 +502,19 @@ public void testDisallowAddListeners() throws Exception {
         }
 
         assertFalse(listeners.addOrNotify(index("1").getTranslogLocation(), new TestLocationListener()));
-        assertFalse(listeners.addOrNotify(index("1").getSeqNo(), new TestSeqNoListener()));
-        assertEquals(3, listeners.pendingCount());
+        final int expectedPending;
+        if (listeners.pendingCount() == maxListeners) {
+            // Rejected
+            TestSeqNoListener rejected = new TestSeqNoListener();
+            assertTrue(listeners.addOrNotify(index("1").getSeqNo(), rejected));
+            assertNotNull(rejected.error);
```

Comment:
Just for my understanding, should `rejected.error` be also asserted in the other branch?